### DISCUSSION TITLE
Metrics : How to get Enh-ACK Probing IE

### DISCUSSION THREAD

**Discussion by amr-ba** [openthread/openthread#8303] [Q&A] [answered]
Hello OpenThread Community

I'm trying to get otLinkMetricsEnhAckProbingIeReportCallback triggered, to receive  Enhanced-ACK Probing, this is what I did : 

1 - Configured Device1 as Leader and Device2 as Child
2 - Device1 :  OtLinkMetricsConfigEnhAckProbing()
3 - Device1 : linkMetricsMgmtResponseCallback was triggered
4 - OtLinkMetricsSendLinkProbe() for iniating probing as mentioned in the Spec, we expect that linkMetricsEnhAckProbingIeReportCallback will be triggered, but it’s not the case,

my questions are: 
How to configure otLinkMetricsEnhAckProbingIeReportCallback ?
Should I use OtLinkMetricsSendLinkProbe for initiating probing? and does it set the Frame Version subfield within the MAC header of the data message to 0b10 ? if it's not the case, how can I set it?

the description in the spec : 
           * The IEEE 802.15.4-2015 Enhanced ACK Probe feature allows the Probing Initiator detailed
           * granular control over link metric queries while saving energy by reusing regular data traffic
           * sequences as an alternative to sending single-purposed MLE Probe messages. The Probing
           * Initiator first configures the Probing Subject via the MLE Probing Management message.
           * Then the Probing Initiator sends regular data messages. Whenever the Probing Initiator
           * would like to query for link metrics, it solicits an Enhanced ACK by setting the Frame
           * Version subfield within the MAC header of the data message to 0b10 and the Probing
           * Subject will respond with an Enhanced Acknowledgement containing link metrics.

Thanks in Advance
Amr

**Comment by Irving-cl** [comment]
Hi Amr,

Thanks for raising the question!

The process you did is correct. If the 'initiator' configures EnhAckProbing successfully at the 'subject', the 'subject' will append a LinkMetrics Report within the Enhanced-ACK sent to the 'initiator'. However, whether replying Enhanced-ACK depends on the incoming data frame. If the version of incoming data frame is 2015, then an Enhanced-ACK will be sent in reply.

Regarding your questions:
> How to configure otLinkMetricsEnhAckProbingIeReportCallback

We should use the API `otLinkMetricsConfigEnhAckProbing` to configure the callback. I think you already did it.

> Should I use OtLinkMetricsSendLinkProbe for initiating probing? and does it set the Frame Version subfield within the MAC header of 
the data message to 0b10 ? if it's not the case, how can I set it?

You can use `otLinkMetricsSendLinkProbe` to trigger a report. But note that, as I mentioned, after an initiator configured EnhAckProbing at a subject, the subject will always append the report within the Enh-ACK whenever the initiator sends **a frame of version 2015**. In OT implementation, we forced all the data frames sent by the initiator to the subject to use version 2015 after EnhAckProbing is configured. See https://github.com/openthread/openthread/blob/main/src/core/thread/mesh_forwarder.cpp#L1841
So any data frame should be ok, including the link probe message.

---
To debug your issue:
* Could you first check the error code return by`linkMetricsMgmtResponseCallback`? Did it succeed?
* What platform (hardware) did you run the OT stack?

**Reply by Irving-cl** [reply]
Update:

Could you also provide the traffic capture of the process?

**Comment by amr-ba** [comment, marked as answer]
Hi,

Sorry for the delay of my answer, 

indeed `linkMetricsMgmtResponseCallback` returned success as status, I found that `OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE` was not enabled and so my leader device is not processing the enhancedAck,

thank you for your help,
Amr
